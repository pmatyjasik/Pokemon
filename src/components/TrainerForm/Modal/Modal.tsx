import styled from "@emotion/styled";
import { Dialog, DialogContent, Button, Typography, Box } from "@mui/material";

const StyledDialog = styled(Dialog)(({ theme }) => ({
    boxShadow: theme.shadows[2],
}));

const StyledDialogContent = styled(DialogContent)(({ theme }) => ({
    backgroundColor: theme.palette.common.white,
    boxShadow: theme.shadows[2],
    padding: theme.spacing(8, 30, 8, 30),
}));

const SuccessMessage = styled(Typography)(({ theme }) => ({
    color: theme.palette.common.black,
    fontSize: theme.spacing(10),
}));

interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
}

export const Modal = ({ isOpen, onClose }: ModalProps) => {
    return (
        <StyledDialog open={isOpen} onClose={onClose}>
            <StyledDialogContent>
                <Box display='flex' flexDirection='column' alignItems='center' justifyContent='center' gap={8}>
                    <SuccessMessage variant='subtitle2' size='lg'>
                        Success
                    </SuccessMessage>
                    <Button onClick={onClose} color='primary' variant='contained'>
                        Reset form
                    </Button>
                </Box>
            </StyledDialogContent>
        </StyledDialog>
    );
};
