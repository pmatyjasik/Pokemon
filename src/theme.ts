"use client";

import { createTheme, Shadows } from "@mui/material/styles";
import { ThemeOptions } from "@mui/material/styles";
import { NextFont } from "next/dist/compiled/@next/font";
import localFont from "next/font/local";

const IbmVga: NextFont = localFont({
    src: "./fonts/IBM_VGA.woff",
    display: "swap",
    style: "normal",
    weight: "400",
});

const themeOptions: ThemeOptions = {
    palette: {
        mode: "light",
        background: {
            default: "rgba(255, 255, 255, 1)",
        },
        primary: {
            main: "rgba(151, 71, 255, 1)",
            dark: "rgba(113, 53, 191, 1)",
            light: "rgba(151, 71, 255, 0.25)",
        },
        grey: {
            100: "rgba(42, 42, 42, 1)",
            200: "rgba(127, 127, 127, 1)",
            300: "rgba(228, 228, 228, 1)",
            400: "rgba(238, 238, 238, 1)",
        },
        error: {
            main: "rgba(255, 78, 78, 1)",
        },
        action: {
            disabled: "rgba(0, 0, 0, 0.2)",
        },
    },
    spacing: 4,
    typography: {
        fontFamily: IbmVga.style.fontFamily,
        fontWeightMedium: 400,
        fontSize: 16,
        h6: {
            fontSize: "12px",
        },
        caption: {
            fontSize: "10px",
        },
    },
    shape: {
        borderRadius: 2,
    },
    shadows: [
        "none",
        "0px 0px 0px 4px rgba(151, 71, 255, 0.25)",
        "0px 4px 10px 2px rgba(0, 0, 0, 0.1)",
        ...Array(20).fill("none"),
    ] as Shadows,
};

export const theme = createTheme(themeOptions);
