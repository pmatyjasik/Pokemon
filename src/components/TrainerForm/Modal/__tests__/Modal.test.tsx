import React from "react";
import { render, screen } from "@testing-library/react";
import { Modal } from "@/components/TrainerForm/Modal/Modal";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "@/theme";

const renderComponent = (props = {}) => {
    return render(
        <ThemeProvider theme={theme}>
            <Modal isOpen={true} onClose={() => {}} {...props} />
        </ThemeProvider>
    );
};

describe("Modal", () => {
    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent();

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders Success message and Reset form button", () => {
        renderComponent();

        expect(screen.getByText("Success")).toBeInTheDocument();
        expect(screen.getByText("Reset form")).toBeInTheDocument();
    });

    test("does not render when isOpen is false", () => {
        renderComponent({ isOpen: false });

        expect(screen.queryByText("Success")).not.toBeInTheDocument();
        expect(screen.queryByText("Reset form")).not.toBeInTheDocument();
    });
});
