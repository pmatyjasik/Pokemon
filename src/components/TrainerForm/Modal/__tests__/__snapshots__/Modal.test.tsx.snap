// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Matches DOM Snapshot 1`] = `<DocumentFragment />`;
